src/smk/tool/geomark/tool-geomark.js

```diff
@@ -211,14 +211,45 @@ include.module( 'tool-geomark', [
             }
 
             this.toGeomark = function(geomarkInfo, drawingLayer) {
+                const geomarkProperties = this.getGeomarkProperties(geomarkInfo);
                 return {
-                    id: geomarkInfo.id || geomarkInfo.properties.id,
-                    url: geomarkInfo.url || geomarkInfo.properties.url,
+                    id: geomarkInfo.id || geomarkProperties.id,
```

Comment:
It looks like geomarkInfo.id and geomarkInfo.url are always going to be undefined.

Reply:
They'll be undefined now and going forward since Geomarks API isn't expected to revert. However at some point these must have had a value (I can't imagine why we'd have added it otherwise). 

I don't yet know whether a Geomark created in an earlier API version will give the same response on GET in a newer API version as it did for the earlier API version. I suspect the response will always follow the current API regardless of the API version it was created with but if it doesn't then keeping this backwards compatibility will be important.

Do you mind if we keep this as-is until we can confirm Geomark API behaviour on this?

Reply:
Update: We also use this method (toGeomark) when creating a Geomark, and in that case, geomarkInfo.id and geomarkInfo.url have a value.

Reply:
Another update: I moved the call to showCouldNotLoadMessage() to the last part of these "chained ORs". I saw that when creating a Geomark, we'd call getGeomarkProperties() and the properties would not be found in the deeper structures where we look for them (because they're at the root in that situation) and we'd incorrectly show the message.